src/game.js

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import {makeShuffledDeck} from "./utils"
+
+class Game extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cardDeck: makeShuffledDeck(),
+      currentCards: [],
+      numOfGames: this.props.games,
+      gameStarted: false,
+      currentRound: 0,
+      currentGame: 1,
+      roundWinner: null,
+      player1CurrScore: 0,
+      player2CurrScore: 0,
+      player1TotalScore: 0,
+      player2TotalScore: 0,
+      
+    };
+  }
+
+  dealCards = () => {
+    const currCards = this.state.cardDeck.slice(-2);
+    if(currCards[0].rank > currCards[1].rank){
+      this.setState({
+        roundWinner: 1,
+        player1CurrScore: this.state.player1CurrScore + 1,
+        player1TotalScore: this.state.player1TotalScore + 1,
+      });
+    }
+    else if (currCards[0].rank < currCards[1].rank){
+      this.setState({
+        roundWinner: 2,
+        player2CurrScore: this.state.player2CurrScore + 1,
+        player2TotalScore: this.state.player2TotalScore + 1,
+      });
+    }
+    else{
+      this.setState({ roundWinner: 0 });
+    }
+      this.setState(
+        (prevState) => ({
```

Comment:
why use the previous state here and not the current state? Any reason?

Reply:
Yeah, this is where my understanding of React setState comes into play. I don't get the full sequence of it. When I was trying to implement a gamesWon tracker, when I tried set it if currentRound >=25, it will not track the game, so I assumed I have to use prevState cause I'm relying on the updated state to see if currentRound >=25. But still does not matter and I couldn't debug it.